### PR TITLE
Fix memory leak in rasm2 by moving ownership of syscall from RAnal to RAsm

### PR DESCRIPTION
**Pull request by MaskRay**

### CODE REVIEW

File: libr/asm/asm.c [hunk 1]
```diff
@@ -198,6 +198,9 @@ R_API RAsm *r_asm_free(RAsm *a) {
 			r_list_free (a->plugins);
 			a->plugins = NULL;
 		}
+		if (a->syscall) {
```
Comment (radare) [inline libr/asm/asm.c]: unnecessary if

Comment (MaskRay) [reply libr/asm/asm.c]: `a->syscall` may be `NULL`

The duplicate of `RAsm::syscall` and `RAnal::syscall` looks suspicious.

**Comment by radare** [comment]
and? freeing NULL should be always fine

File: libr/asm/asm.c (hunk 1 above)
Comment (radare) [reply libr/asm/asm.c]: there's no problem on calling r_syscall_free(NULL); so remove the if

**Comment by radare** [comment]
done in master

**Pull request closed without merging**